Workbooks/AKS/NPM Configuration/NPM Configuration.workbook

```diff
@@ -0,0 +1,604 @@
+                    "options": {
+                      "style": "decimal"
+                    }
+                  }
+                }
+              }
+            },
+            "customWidth": "50",
+            "showPin": true,
+            "name": "Num IPSets",
+            "styleSettings": {
+              "showBorder": true
+            }
+          }
+        ]
+      },
+      "conditionalVisibility": {
+        "parameterName": "selectedTab",
+        "comparison": "isEqualTo",
+        "value": "Summary Counts"
+      },
+      "name": "Summary Counts"
+    },
+    {
+      "type": 12,
+      "content": {
+        "version": "NotebookGroup/1.0",
+        "groupType": "editable",
+        "items": [
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "InsightsMetrics\r\n| where Namespace == \"prometheus\"\r\n| where Name == \"npm_num_ipset_entries\"\r\n| summarize arg_max(TimeGenerated, Val)",
+              "size": 4,
+              "title": "Total Entries",
+              "timeContext": {
+                "durationMs": 86400000
+              },
+              "queryType": 0,
+              "resourceType": "microsoft.operationalinsights/workspaces",
+              "visualization": "tiles",
+              "tileSettings": {
+                "titleContent": {
+                  "formatter": 1
+                },
+                "secondaryContent": {
+                  "columnMatch": "Val",
+                  "formatter": 8,
+                  "formatOptions": {
+                    "max": 60000,
+                    "palette": "orange"
+                  }
+                },
+                "showBorder": false,
+                "rowLimit": 1,
+                "size": "auto"
+              },
+              "graphSettings": {
+                "type": 0
+              },
+              "mapSettings": {
+                "locInfo": "LatLong",
+                "sizeSettings": "sum_Val",
+                "sizeAggregation": "Sum",
+                "legendMetric": "sum_Val",
+                "legendAggregation": "Sum",
+                "itemColorSettings": {
+                  "type": "heatmap",
+                  "colorAggregation": "Sum",
+                  "nodeColorField": "sum_Val",
+                  "heatmapPalette": "greenRed"
+                }
+              }
+            },
+            "customWidth": "15",
+            "name": "Total IPSet Entries"
+          },
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "InsightsMetrics\r\n| where Namespace == \"prometheus\"\r\n| where Name == \"npm_ipset_counts\"\r\n| where TimeGenerated > ago(2m)\r\n| extend tags = parse_json(Tags)\r\n| extend setName = tostring(tags.set_name)\r\n| summarize arg_max(TimeGenerated, Val) by setName\r\n| where Val > 0\r\n| order by Val desc, setName asc\r\n| project Set = setName, Entries = Val",
```

Comment:
@vishiy this query depends upon the scrape interval for the advanced metric with high cardinality. Should I increase the time in the "ago" clause? Our NPM deletes the metrics with 0 counts, so if we don't include an ago clause, the grid will incorrectly display the last-scraped old counts for any deleted ipset.

Reply:
We will scrape all with a 1 min frequency. But rather then hard coding time, you should pass in as a parameter (thru a time selector in the workbook). Just take a look at our other workbooks in a cluster.